program/lib/Roundcube/rcube_imap_generic.php

```diff
@@ -3049,21 +3049,12 @@ protected static function decodeContent($chunk, $mode, $is_last = false, &$prev
     {
         // BASE64
         if ($mode == 1) {
-            $chunk = $prev . preg_replace('|[^a-zA-Z0-9+=/]|', '', $chunk);
-
-            // create chunks with proper length for base64 decoding
-            $length = strlen($chunk);
-
-            if ($length % 4) {
-                $length = floor($length / 4) * 4;
-                $prev = substr($chunk, $length);
-                $chunk = substr($chunk, 0, $length);
+            $decoded_chunk = '';
+            foreach(preg_split("/((\r?\n)|(\r\n?))/", $chunk) as $line){
```

Comment:
```suggestion
            foreach(preg_split('~[\s=]+~', $chunk) as $line) {
```

seems to work well https://3v4l.org/5uMok - a few edge cases should be added to unit tests